spring-pulsar-reactive/build.gradle

```diff
@@ -17,7 +25,6 @@ dependencies {
 	optional 'com.google.protobuf:protobuf-java'
 	optional 'com.jayway.jsonpath:json-path'
 	optional 'io.projectreactor:reactor-core'
-	optional 'org.apache.pulsar:pulsar-client-reactive-producer-cache-caffeine'
```

Comment:
Now that Caffeine is shaded in the cache module we can bring it in by default as it is recommended to use the cache module. This is the same thing we do for the imperative side (it brings in shaded cache module as well).